lib/jake/string.ex

```diff
@@ -1,21 +1,46 @@
 defmodule Jake.String do
-  def gen(spec) do
-    options = []
-
-    options =
-      if min_length = Map.get(spec, "minLength") do
-        Keyword.put(options, :min_length, min_length)
-      else
-        options
-      end
-
-    options =
-      if max_length = Map.get(spec, "maxLength") do
-        Keyword.put(options, :max_length, max_length)
-      else
-        options
-      end
-
-    StreamData.string(:ascii, options)
+  @strlen_min 1
+
+  @strlen_max 100
+
+  def gen_string(map), do: stringer(map, map["enum"], map["pattern"])
+
+  def find_min_max(map) do
+    min = Map.get(map, "minLength", @strlen_min)
+    max = Map.get(map, "maxLength", @strlen_max)
+    {min, max}
+  end
+
+  def stringer(map, enum, pattern) when is_nil(enum) and is_nil(pattern) do
+    {min, max} = find_min_max(map)
+    re = Randex.stream(~r/[a-zA-Z0-9\_]{#{min},#{max}}/)
+
+    if min <= max do
+      StreamData.bind_filter(StreamData.string(:alphanumeric), fn
```

Comment:
why not use min_length/max_length instead of bind_filter?

Reply:
It was written when I did not read StreamData.string method properly. Later on I kept it, as it is just to know your opinion on the above manual method.